pkg/tasks/prometheus.go

```diff
@@ -299,6 +299,18 @@ func (t *PrometheusTask) Run() error {
 			return errors.Wrap(err, "syncing Prometheus trusted CA bundle ConfigMap failed")
 		}
 
+		secret, err := t.factory.AdditionalAlertManagerConfigsSecret()
+		if err != nil {
+			return errors.Wrap(err, "initializing Prometheus additionalAlertManagerConfigs secret failed")
+		}
+		if secret != nil {
+			klog.V(4).Info("initializing Prometheus additionalAlertManagerConfigs secret")
+			err = t.client.CreateOrUpdateSecret(secret)
+			if err != nil {
+				return errors.Wrap(err, "reconciling Prometheus additionalAlertManagerConfigs secret failed")
+			}
+		}
```

Comment:
Late realization for me (sorry!) but we need to delete the secret in case the admin had configured additional Alertmanagers and then remove them all.